tests/flows/check_upgrade_access.go

```diff
@@ -0,0 +1,86 @@
+package flows
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/accounts/abi/bind"
+	"github.com/ava-labs/teleporter/tests/interfaces"
+	"github.com/ava-labs/teleporter/tests/utils"
+	"github.com/ethereum/go-ethereum/crypto"
+	. "github.com/onsi/gomega"
+)
+
+func CheckUpgradeAccess(network interfaces.Network) {
+	subnetInfo := network.GetPrimaryNetworkInfo()
+	_, fundedKey := network.GetFundedAccountInfo()
+
+	//
+	// Deploy ExampleMessenger to Subnets A and B
+	//
+	ctx := context.Background()
+	teleporterAddress := network.GetTeleporterContractAddress()
+	_, exampleMessenger := utils.DeployExampleCrossChainMessenger(
+		ctx, fundedKey, subnetInfo,
+	)
+
+	// Try to call updateMinTeleporterVersion from a non owner account
+	nonOwnerKey, err := crypto.GenerateKey()
+	Expect(err).Should(BeNil())
+	nonOwnerAddress := crypto.PubkeyToAddress(nonOwnerKey.PublicKey)
+
+	// Transfer native assets to the non owner account
+	fundAmount := big.NewInt(0).Mul(big.NewInt(1e18), big.NewInt(1)) // 10eth
+	fundNonOwnerTx := utils.CreateNativeTransferTransaction(
+		ctx,
+		subnetInfo,
+		fundedKey,
+		nonOwnerAddress,
+		fundAmount)
+	utils.SendTransactionAndWaitForSuccess(ctx, subnetInfo, fundNonOwnerTx)
+
+	nonOwnerOpts, err := bind.NewKeyedTransactorWithChainID(
+		nonOwnerKey, subnetInfo.EVMChainID)
+	Expect(err).Should(BeNil())
+	ownerOpts, err := bind.NewKeyedTransactorWithChainID(
+		fundedKey, subnetInfo.EVMChainID)
+	Expect(err).Should(BeNil())
```

Comment:
nit: I'd prefer we grouped the `nonOwnerOpts` with `_, err = exampleMessenger.PauseTeleporterAddress(nonOwnerOpts, teleporterAddress)` and then `ownerOpts` with the call that uses them.

ie delcaring something closest to where it is being used